File: codechef/START84C/6_DIGITOP.py
from bisect import bisect, bisect_left
t = int(input())

for _ in range(t):
    n,k = map(int,input().split())
    a = input().split(" ")
    b = input().split(" ")

    flag = True
    for i in range(len(a)):
        if len(a[i]) != len(b[i]):
            flag = False
            break

    a = sorted("".join(a))
    b = sorted("".join(b))
    c = ['0','1','2','3','4','5','6','7','8','9']
    positive = 0
    negative = 0
    result = 0
    for i in c:
        counta = bisect(a,i) - bisect_left(a,i)
        countb = bisect(b,i) - bisect_left(b,i)
        if counta <  countb:
            negative += countb - counta
        elif counta > countb:
            positive += counta - countb
    if negative > positive:
        result = negative - positive
    else:
        result = negative

    if flag and result <=k:
        print("YES")
    else:
        print("NO")

File: codechef/START84C/6_DIGITOP.cpp
#include<iostream>
#include<algorithm>
#include<string>

#define ios_base ::sync_with_stdio(false); cin.tie(0); cout.tie(0);

using namespace std;


int main(){

    int t;
    freopen("6_DIGITOP_input.txt","r",stdin);
    cin>>t;
    while(t--){
        int n;
        long long k;
        string temp;
        cin>>n>>k;
        string a,b;
        getline(cin, a);
        getline(cin, b);
        bool flag = true;
        for (int i=0;i<a.size();i++)
            if ((a[i] == ' ' && b[i] != ' ') || (a[i] != ' ' && b[i] == ' ')) {flag = false; break;}
        sort(a.begin(),a.end());
        sort(b.begin(),b.end());
        long long positive = 0, negative = 0, result = 0;
        for (char i='0';i<='9';i++){
            int counta = upper_bound(a.begin(),a.end(),i) - lower_bound(a.begin(),a.end(),i);
            int countb = upper_bound(b.begin(),b.end(),i) - lower_bound(b.begin(),b.end(),i);
            if (counta > countb) positive += counta-countb;
            else if (counta<countb) negative += countb-counta;
        }
        if (negative > positive) 
            result = negative;
        else result = negative + (positive - negative);
        if (flag && result <= k) cout<<"YES"<<"\n";
        else cout<<"NO"<<"\n";
    }
    return 0;
}